### ISSUE TITLE
[packaging] get Seafile Server into Debian

### ISSUE THREAD

**Issue by hex-m** [haiwen/seafile#1916]
As the [licensing issues](https://github.com/haiwen/seafile/issues/666) seem resolved, it would be great to get the server parts of Seafile into Debian.

From a "competitive" perspective there is a void to be filled as OwnCloud was dropped from Debian.

**Comment by christianfl** [comment]
Please have a look here:
https://forum.seafile.com/t/what-about-reproducible-builds/3125/11

**Comment by hex-m** [comment]
Thanks for the link! The relevant parts:

> The server won't be officially packaged for Debian.
> The development cycle for the server isn't compatible with the release philosophy of Debian - maintaining such a package would probably be a living hell of backporting fixes while fending of annoyed users that want to have new features...
So we agreed that it's best to leave the server for self-installing.

**Comment by moschlar** [comment]
See also: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=865830#25

**Comment by niol** [comment]
Any effort to get seafile-server will be blocked by the libzdb copyright issue raised a while ago. See a description of the problem here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=928975#5

**Comment by killing** [comment]
The code that looks similar to libzdb will be replaced to avoid any confusion.

**Comment by hex-m** [comment]
> The code that looks similar to libzdb will be replaced to avoid any confusion.

Does the closing of the issue mean, that this happened? Can you link to the PRs?

Or does this mean `wontfix`?

**Comment by killing** [comment]
I would say this is wontfix. It's hard to package the server into debian package.

**Comment by niol** [comment]
No it's not. I've done a while ago a proof of concept for 6.2 that is documented : https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=865830#25

There are two kinds of problems:
- legal problems regarding licensing which I think have been fixed
- technical issues regarding embedded dependecies and conforming to the unix way of where to put files, and fixes for those have been in pull requests for a while, for instance https://github.com/haiwen/seafile-server/pull/215 https://github.com/haiwen/seafile-server/pull/161 https://github.com/haiwen/seafile-server/pull/119 https://github.com/haiwen/seafile-server/pull/109 https://github.com/haiwen/seafile-server/pull/97 https://github.com/haiwen/ccnet-server/pull/43 https://github.com/haiwen/ccnet-server/pull/25 https://github.com/haiwen/ccnet-server/pull/10, which has discouraged me of proposing more, for instance for seahub.

Integrating this patches would motivate me to pursue my effort. I think seafile is great and I've happilly used my 6.2 packages for 2 years now. But debian packages bring in more security (because no embedded deps, read only seahub, etc.) and easier installation.

**Comment by hoxu** [comment]
I'm personally really disappointed if all efforts to get seafile server into Debian are abandoned.
